File: PianoRiseDotnet/PianoRiseApp/Models/Historial.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PianoRiseManagement.Models
{
    [Table("historial")]
    public class Historial
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("alumno_id"), ForeignKey("Alumno")]
        public int AlumnoId { get; set; }

        [Column("aula_id"), ForeignKey("Aula")]
        public int AulaId { get; set; }

        [Column("cancion_id"), ForeignKey("Cancion")]
        public int CancionId { get; set; }

        [Column("precision")]
        public decimal Precision { get; set; }

        [Column("errores")]
        public int Errores { get; set; }

        [Column("duracion")]
        public TimeSpan Duracion { get; set; }

        [Column("registrado_en")]
        public DateTime RegistradoEn { get; set; }

        public Usuario Alumno  { get; set; } = null!;
        public Aula    Aula    { get; set; } = null!;
        public Cancion Cancion { get; set; } = null!;
    }
}

File: PianoRisePython/app/models/historial.py
from datetime import datetime
from sqlalchemy import Column, Integer, Numeric, DateTime, ForeignKey, Interval
from sqlalchemy.orm import relationship
from app import db

class Historial(db.Model):
    __tablename__ = "historial"

    id            = Column("id", Integer, primary_key=True)
    alumno_id     = Column("alumno_id", Integer, ForeignKey("usuario.id"),    nullable=False)
    aula_id       = Column("aula_id",   Integer, ForeignKey("aula.id"),       nullable=False)
    cancion_id    = Column("cancion_id",Integer, ForeignKey("cancion.id"),   nullable=False)
    precision     = Column("precision", Numeric, nullable=False)
    errores       = Column("errores",   Integer, nullable=False)
    duracion      = Column("duracion",  Interval, nullable=False)
    registrado_en = Column("registrado_en", DateTime, default=datetime.utcnow, nullable=False)

    usuario  = relationship("Usuario",   back_populates="historiales")
    aula     = relationship("Aula",      back_populates="historiales")
    cancion  = relationship("Cancion",   back_populates="historiales")
